File: LeetCode/Easy/219 Contains Duplicate II.js
// forEach((value, idx) => {...}) 은 중간에 break 시킬 수 없다.

var containsNearbyDuplicate = function(nums, k) {
    var tmp = new Map();
    for (let idx = 0; idx < nums.length; idx++) {
        let num = nums[idx];
        if (tmp.has(num) && (idx - tmp.get(num) <= k)) {
            return true;
        }
        tmp.set(num, idx);
    }
    return false;
};

File: LeetCode/Easy/219 Contains Duplicate II.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        dic = dict()
        for idx, num in enumerate(nums):
            if num in dic and idx - dic[num] <= k:
                return True
            dic[num] = idx
        return False